cmd/tg_bot/internal/environment.go

```diff
@@ -154,3 +162,124 @@ func (tgEnv *TelegramBotEnvironment) NodesListMessage(urls []string) (string, er
 	}
 	return w.String(), nil
 }
+
+func (tgEnv *TelegramBotEnvironment) SubscribeToAllAlerts() error {
+	for alertType := range entities.AlertTypes {
+		alertName, ok := entities.AlertTypes[alertType] // check if such an alert exists
+		if !ok {
+			return errors.New("failed to subscribe to alerts, unknown alert type")
+		}
+
+		err := tgEnv.pubSubSocket.SetOption(mangos.OptionSubscribe, []byte{byte(alertType)})
+		if err != nil {
+			return err
+		}
+		tgEnv.subscriptions[alertType] = alertName
+		log.Printf("Subscribed to %s", alertName)
+	}
+
+	return nil
+}
+
+func (tgEnv *TelegramBotEnvironment) SubscribeToAlert(alertType entities.AlertType) error {
+	alertName, ok := entities.AlertTypes[alertType] // check if such an alert exists
+	if !ok {
+		return errors.New("failed to subscribe to alert, unkown alert type")
+	}
+
+	err := tgEnv.pubSubSocket.SetOption(mangos.OptionSubscribe, []byte{byte(alertType)})
+	if err != nil {
+		return errors.Wrap(err, "failed to subscribe to alert")
+	}
+
+	tgEnv.subscriptions[alertType] = alertName
+	log.Printf("Subscribed to %s", alertName)
+	return nil
+}
+
+func (tgEnv *TelegramBotEnvironment) UnubscribeFromAlert(alertType entities.AlertType) error {
+	alertName, ok := entities.AlertTypes[alertType] // check if such an alert exists
+	if !ok {
+		return errors.New("failed to unsubscribe from alert, unkown alert type")
+	}
+	err := tgEnv.pubSubSocket.SetOption(mangos.OptionUnsubscribe, []byte{byte(alertType)})
+	if err != nil {
+		return errors.Wrap(err, "failed to unsubscribe from alert")
+	}
```

Comment:
What if you already has been unsubscribed from this alert? I think the `if` below should be placed above `SetOption` call.